### ISSUE TITLE
Custom bindings for `Confirm`

### ISSUE THREAD

**Issue by arcanis** [enquirer/enquirer#313]
Currently Enquirer makes it easy enough to do things like:

```
Stage this hunk [y,n]?
```

However, in various context (for instance when running `git add -p`), you want to do:

```
Stage this hunk [y,n,q,a,d,e,?]?
```

From an API perspective, ideally this would look something like this:

```ts
const res = await Enquirer.prompt([{
  type: `confirm`,
  name: `confirm`,
  message: `Stage this hunk`,
  extra: [`q`, `a`, `d`, `e`, `?`],
}]);

switch (res) {
  case true: ...
  case false: ...
  case `q`: ...
  ...
}
```

**Comment by doowb** [comment]
Hi @arcanis,

TL;DR; This is possible by passing an `isTrue` function on the options:

```js

const { prompt } = require('enquirer');

(async () => {
  const choices = [`q`, `a`, `d`, `e`, `?`];

  const { confirm } = await prompt([{
    type: `confirm`,
    name: `confirm`,
    message: `Stage this hunk`,
    default: `[y,n,${choices.join(',')}]`,
    isTrue(input) {
      return /^[ty1]/i.test(input) ? true : (choices.find(ele => ele === input) ? input : false);
    }
  }]);

  console.log(confirm);

  switch (confirm) {
    case true: {
      console.log('TRUE');
      break;
    }
    case false: {
      console.log('FALSE');
      break;
    }
    case `q`: {
      console.log('Q');
      break;
    }
    case `a`: {
      console.log('A');
      break;
    }
    case `d`: {
      console.log('D');
      break;
    }
    case `e`: {
      console.log('E');
      break;
    }
    case `?`: {
      console.log('?');
      break;
    }
    default: {
      console.log('UNKNOWN');
    }
  }
})();
```

Notice that you have to set the `default` option too if you want the list of choices to display correctly. Doing it this way will allow returning a result as one of the following `true, false, 'q', 'a', 'd', 'e', '?'`. This still allows typing `'t', 'y', or 1` for `true` and `'f', 'n', or 0` for `false`.

The 'Select' prompt might be a better choice because it's made for selecting a single choice from a list of choices. I think it would provide a more complete solution, but would require handling keypresses (see the [toggle prompt](https://github.com/enquirer/enquirer/blob/master/lib/prompts/toggle.js#L58-L74) for example) and rendering the choices correctly.
